sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -376,12 +376,12 @@ abstract class GpuSpecifiedWindowFrameMetaBase(
           return None
         }
 
-        val value: Long = bounds match {
+        val value: BigInt = bounds match {
```

Comment:
Since this returns `BigInt`, all the values below should be casted to that type.

Reply:
The snag is that there is no `.toBigInt`, like there is a `.toLong`.

However, integral types have an implicit conversion to `BigInt`, so an explicit cast is unnecessary.

Reply:
Note: The following is the verbose alternative:
```scala
        val value: BigInt = bounds match {
          case Literal(value, ByteType) => BigInt(value.asInstanceOf[Byte])
          case Literal(value, ShortType) => BigInt(value.asInstanceOf[Short])
          case Literal(value, IntegerType) => BigInt(value.asInstanceOf[Int])
          case Literal(value, LongType) => BigInt(value.asInstanceOf[Long])
          case Literal(value: Decimal, DecimalType()) => value.toJavaBigDecimal.unscaledValue()
          case Literal(ci: CalendarInterval, CalendarIntervalType) =>
            if (ci.months != 0) {
              willNotWorkOnGpu("interval months isn't supported")
            }
            // return the total microseconds
            try {
              BigInt(Math.addExact(
                Math.multiplyExact(ci.days.toLong, TimeUnit.DAYS.toMicros(1)),
                ci.microseconds))
            } catch {
              case _: ArithmeticException =>
                willNotWorkOnGpu("windows over timestamps are converted to microseconds " +
                  s"and $ci is too large to fit")
                if (isLower) BigInt(-1) else BigInt(1) // not check again
            }
          case _ => BigInt(getAndTagOtherTypesForRangeFrame(bounds, isLower))
        }

```

Reply:
I am fine with the current implementation. I am not thrilled about value becoming a BigInt, as that is a class and can add overhead, but this happens rarely enough that I don't think we need to spend a bunch of time optimizing it.  From a readability standpoint I don't think either really improves it much.

Reply:
Okay then I'm fine with it.